src/server/models/list.ts

```diff
@@ -42,6 +44,78 @@ export async function findListByCountryAndType(country: CountryName, type: Servi
   }
 }
 
+export async function findListByAnnualReviewDate(annualReviewStartDate: Date, listItemStatuses?: Status[], isAnnualReview?: boolean): Promise<Result<List[]>> {
+  try {
+    logger.debug(`searching for lists matching date [${annualReviewStartDate}]`);
+
+    const result = (await prisma.list.findMany({
+      where: {
+        nextAnnualReviewStartDate: annualReviewStartDate,
+        isAnnualReview: false,
+      },
+      include: {
+        country: true,
+        items: {
+          where: {
+            list: {
+              jsonData: {
+                path: ["currentAnnualReview", "eligibleListItems"],
+                array_contains: [],
+              },
+            },
+            ...(listItemStatuses != null && { status: { in: listItemStatuses }}),
```

Comment:
non strict equality or inequality (like == or !==) should be fine, but worth being aware why it works. (And I'm surprised eslint doesn't moan at you for this).

```
console.log(undefined === null) // false
console.log(undefined == null) // true
```

for non-strict (in)equality, if the two values are different types, they'll be casted and compared. eslint rules are typically defaulted to not allow non-strict (in) equalities because of the confusing rules/behaviour around it.

I don't think `!= null` is necessary and confuses matters. The parameter is `listItemStatuses?: Status[]`. For ts, `?` type definition means undefined. i.e., `listItemStatuses` will never be a null value judging by the type definition. It can be an undefined value. 

if nothing is passed into a parameter, the parameter is undefined (and not null). 
```
findListByAnnualReviewDate(Date.now())) 
// listItemStatuses === undefined is true.
```

for listItemStatuses, I think it's best to do a length check on the array, in case an empty array is passed, which I assume would give you no or all results (unsure of prisma behaviour. i.e. query is filtering by statuses that match the array `[]`)
`findListByAnnualReviewDate(annualReviewStartDate: Date, listItemStatuses?: Status[] = [], isAnnualReview?: boolean)` (default listItemStatuses to empty array)

isAnnualReview can stay as it is, but undefined (vs null) check would make more sense. (i.e. you don't want to default this to true or false, since it will only find isAnnualReview === false or === true)

Reply:
yeah defaulting definitely makes more sense here.  Not sure why I chose != null.  Ta